sycl/test/extensions/free_function_errors.cpp

```diff
@@ -25,6 +25,10 @@ union U {
 
 using accType = accessor<int, 1, access::mode::read_write>;
 
+struct Sacc {
+  accType acc;
+};
+
```

Comment:
Please add this case to clang/test/SemaSYCL tests instead. The fact that we are checking diagnostics in sycl/tests is a mistake that we need to get rid of, not extend.

Reply:
Turns out we already test this case in `clang/test/SemaSYCL/free_function_kernel_param_restrictions` so I just removed it here.